### DISCUSSION TITLE
Is there any way to receive 2 or more abritrary values?

### DISCUSSION THREAD

**Discussion by scv057** [tailwindlabs/tailwindcss#11779] [Help] [answered]
I want to create a class that can receive mutiple abritrary values like `my-class-{v1}-{v2}`, how can I achieve it?

I tried `matchUtilities` seems only can receive one value.

Can anybody help me? many thanks🙏🏻🙏🏻🙏🏻

**Comment by wongjn** [comment, marked as answer]
Have some kind of predetermined delimiter in the arbitrary value and then split the value by this delimiter.

For example, here we use a plus sign (`+`) as a delimiter:

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  plugins: [
    require('tailwindcss/plugin')(({ matchUtilities }) => {
      matchUtilities({
        'my-class': value => Object.fromEntries(
          value.split('+').map((v, i) => [`--foo-${i}`, v])
        ),
      });
    })
  ],
}
```

Then you'd use arbitrary values like `my-class-[a+b+c]`.

See [this Tailwind Play](https://play.tailwindcss.com/IAfb5dbaxn?file=config) for a working example (open the *Generated CSS* panel to see the CSS results).

**Reply by scv057** [reply]
Thank you very much!!

**Reply by kingscode-big** [reply]
thanks ,,this has reaally helped me

**Comment by orimay** [comment]
Is there a way to do so in TailwindCSS v4?

**Reply by wongjn** [reply]
You can support up to 2 values using `--value()` and `--modifier()`. See https://tailwindcss.com/docs/adding-custom-styles#modifiers for details. Otherwise, if you need more values, you'd need to split them up into separate class names and use CSS variables to "compose" them together.

**Reply by orimay** [reply]
Yeah, this get very ugly. I need to support colors with arbitrary hue shifts and opacity

**Reply by orimay** [reply]
Wouldn't have been much of an issue if I could use SCSS to generate Tailwind config
